pkg/daemon/ceph/osd/volume.go

```diff
@@ -79,9 +80,26 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		strconv.Itoa(a.storeConfig.OSDsPerDevice),
 	}...)
 
-	// ceph-volume is soon implementing a parameter to specify the "fast devices", which correspond to the "metadataDevice" from the
-	// crd spec. After that is implemented, we can implement this. In the meantime, we fall back to use rook's partitioning.
+	if a.storeConfig.StoreType == config.Bluestore && a.storeConfig.DatabaseSizeMB > 0 {
+		if a.storeConfig.DatabaseSizeMB < 1024 {
+			// ceph-volume will convert this value to ?G. It needs to be > 1G to invoke lvcreate.
+			logger.Infof("skipping databaseSizeMB setting. For it should be larger than 1024MB.")
+		} else {
+			batchArgs = append(batchArgs, []string{
+				databaseSizeFlag,
+				// ceph-volume takes in this value in bytes
+				strconv.FormatInt(int64(a.storeConfig.DatabaseSizeMB)*1024*1024, 10),
+			}...)
+		}
+	}
+
+	// When mixed hdd/ssd devices are given, ceph-volume configures db lv on the ssd.
 	metadataDeviceSpecified := false
+	if a.metadataDevice != "" {
+		logger.Infof("using %s as metadataDevice and let ceph-volume lvm batch decide how to create volumes", a.metadataDevice)
```

Comment:
Why don't you use `--report` to print what c-v is going to do?

Reply:
We have `--report` in just right before the `--prepare`. It's a better place considering the code structure.

Reply:
My bad, I missed that.